### ISSUE TITLE
Auto sharing folder on wizard

### ISSUE THREAD

**Issue by onon765trb** [DigitalMediaServer/DigitalMediaServer#101]
DMS automatically share Desktop and Downloads folders.
I think this is not good on security and the burdens of PC.
Desktop and Downloads folders are privacy itself.
It is a cause for unexpected outflows by beginners.

And I think agreements by users are need, including what folders is shared, when automatically sharing folders,

**Comment by Nadahar** [comment]
What to share by default is a tricky decision. I don't expect most users to use the default shares at all, but it's important that something is shared for people that are completely new to the software. When people install it for the first time and isn't familiar with media servers, there can be a lot to learn/understand to get things working. The idea is that *something* should be shared so that they can find it on their renderers when the network setup and other things work as it should.

UMS simply shares all local hard-drives by default, *that* is what I'd call a privacy issue. That said, DMS only shares audio, video and image files, so things like documents, certificates and other potentially privacy sensitive file types aren't shared in any case. In addition, anything shared by DMS is only shared on the *local* network.

How people use the "standard folders" probably varies a lot. I would never put anything remotely sensitive in either of these folders, but I do realize that other users' habits might be very different.

Why do you think that Desktop and Downloads are that sensitive?

**Comment by SubJunk** [comment]
@Nadahar we share only Audio, Video and Music directories by default since the last version.

It's funny because straight after that release we got complaints from users that they couldn't see all their hard drives anymore, and it is a burden to add each drive manually :laughing: no matter which decisions we make we upset some people

**Comment by Nadahar** [comment]
@SubJunk Yes, it just shows how difficult it is to get the default right.

**Comment by onon765trb** [comment]
Many people use Desktop and Downloads folders as temporal work space because they are easy to use. Management of too many files by folders become difficult in the present.
As a result, Apple enhances "spotlight" or "stack" of Desktop. 

Surely, UMS have share only Audio, Video, Picture and Music directories by default since my pointing out in ISSUE [1572](https://github.com/UniversalMediaServer/UniversalMediaServer/pull/1572) or  in ISSUE [1574](https://github.com/UniversalMediaServer/UniversalMediaServer/pull/1574).
But my comments cannot be looked though I do not know reasons.

At least, veteran users warned this for beginners in Japanese bbs. And I never have found OSs share these folders by default.
And it is I who do not do Self-made performance or act another person.

I hope @Nadahar is wise for safely enjoy.

**Comment by Nadahar** [comment]
I don't know where your comments have gone either. As you sure you commented in one of those? If you comment directly in the code and the code is later changed, those comments will disappear - but comments made in the issue or pull request itself shouldn't disappear unless somebody deletes them. If somebody deletes them, GitHub will log that so you can see that comments were deleted even if you can't see the comments.

To make sure that you don't misunderstand me, I'm not against removing these from the default shared folders. I just want to make sure that we make the correct decision, which is why I want to discuss it at bit first.

I don't think OS shared folders are directly comparable, since they share all files, not only media files. Maybe I don't understand what you mean by "privacy". For me, privacy is about personal information, passwords, bank information, political views, health information, personal letters, digital certificates etc. I don't quite understand how this type of information would be available in media files in the first place. I guess there could be scanned documents that were stored as images that could be shared, but I would find it very strange if people store these on the desktop.

I'm wondering if you're thinking of something very different than me when you talk about privacy. Since I'm not Japanese and not a Mac user, there might be things that's obvious too you that I will never think of. Can you give me an idea of what kind of media files that would be stored on the desktop or in the `Downloads` folder that would be a privacy risk if shared on the local network? In most cases, the same people that can access the local network can also access the computer itself.

I will try to explain why I included these folders in the first place. From my experience with working in IT I have learned that, even though it's hard to understand, there are many people using computers that don't understand file systems and the concept of files and folders.

These people use applications and always accept the default locations. They have no idea that they can move files or create folders and don't quite understand the concept. They often think that the documents "belong" to the application, for example that PSD's "live" in Photoshop. If they save a PSD they will give it a name and store it in the folder suggested by Photoshop, and when they want to open it they will look for the same name they used when they stored it. They don't really understand the file concept, so they just think of what they see in the file dialog as some kind of "project index" for that application.

If these people download a media file from the internet to test with DMS, it will be probably be found in the `Downloads` folder. In the same way if they transfer media from their camera or mobile phone, these media will be where the "transfer application" puts them. That might be in the appropriate media folder, the desktop or somewhere else, all depending on the "transfer application".

These people, in my experience, tend to think of the computer as a somewhat abstract concept of a "black box" where things exist inside. That means that if they have transferred images or videos from their camera to the computer, or downloaded media files, they will expect that these files will be available for DMS to share.

While I don't understand how people can use computers without understanding file systems, I would estimate that this applies to somewhere between 20% and 30% in a typical workplace. I therefore expects that this is also true for a similar percentage of users in general, if not even higher. I simply don't want them to try DMS and conclude that "it doesn't work" or "it's to difficult for me" and then give up. If they get it working (that is, is able to find their media files on their TV, mobile phone etc) from the start, maybe they will get interested enough to take the time to understand a bit more about files and folders.

To me, using a computer without this knowledge is like driving a car without knowing any traffic rules or traffic signs, it must be a quite frustrating experience. I guess it has to do with personality types, I will always investigate things I don't understand. Others have a different philosophy where they try to never learn more than they have to. I guess these are the people that never learn what files and folders are, but it's still very hard for me to understand :wink:

**Comment by SubJunk** [comment]
@onon765trb your comments are in a related issue https://github.com/UniversalMediaServer/UniversalMediaServer/pull/1570#issuecomment-422290074

**Comment by onon765trb** [comment]
@Nadahar
I am glad that I could have know your idea. You are very kind for users. By your idea, I can continue to cooperate. This is because I have plans for extensions, still.

We do not have to forget that high end users recognize unexpected behavior in **users not knowing** as danger. Those who suffers damages are beginners. And beginners follow the high end users. There are disappeared softwares, due to such behavior, regardless of codes or the intension of developers. Some PMS users continue to use PMS or forks of PMS in Japan for some reasons.
I think that security is one of the most important matters. My real intention is to stop auto share. If not so, I have no difficulty because of being able to customize codes.

I understand that the default makes the usage of DLNA easy. I am thinking your kind idea is easily achieved by the check box of share folder tab, like iTunes library.

**Comment by Sami32** [comment]
Copy of my message posted in the wrong thread.
>As a compromise i suggest to add in our last wizard window related to the library folders choice to add a line asking users if they want add the "Desktop" (i'll not do any objection if this one was removed) and the "Download" folder by default.
As users will mostly hit "Yes" without reading the message as we already saw many times, it should be fine to avoid complain that they don't find they media and from the few other wishing not having them added for burding issue as these last ones should take the time to read the message.

**Comment by Sami32** [comment]
@Nadahar Then as i said, just ask to the user in the last window wizard with the default preselected answer on "Yes".

**Comment by Nadahar** [comment]
@Sami32 Yes, I think that would be a reasonable solution too, but I'd like to hear if this would satisfy @onon765trb.

**Comment by onon765trb** [comment]
@Nadahar @Sami32
I agree in great part.
My anxiety is
> As users will mostly hit "Yes" without reading the message

My example
There are 15 thousands files in  "Desktop" folder and 60 thousands ones in  "Download" though these include source files. 

I think many users cannot figure out what files exist there.
Blessedly, DMS open "home" folder when selecting shared folders. It means "Desktop" and "Download" folder appear right in front in split seconds, clicking "Add folder" button.
My present request is to exclude "Desktop" and "Download" folders from the default. "Public", too, if possible. This is because the folder is used in acceptance or delivery of files, rather than opening to many people.
At least, I cannot find (real) complaints that all drives or all files of  "home" folder is not automatically shared.
Rather than it, it is recognized as danger.

**Comment by Sami32** [comment]
@onon765trb I understand that you care for that but:
https://www.universalmediaserver.com/forum/viewtopic.php?f=3&t=12880#p37522
https://www.universalmediaserver.com/forum/memberlist.php?mode=viewprofile&u=2299
https://www.universalmediaserver.com/forum/viewtopic.php?f=3&t=12809&start=10#p37340
https://www.universalmediaserver.com/forum/viewtopic.php?f=10&t=11636

So if the users don't read the message we cannot be hold as responsible for that, as the insurers and bankers told us when we didn't read the little lines before read ;)
How many of us have read the Google little lines? only 150 pages that care about our privacy i think ;)

**Comment by Nadahar** [comment]
@onon765trb I really, really don't understand why anyone would have 15 000 files on their desktop and 60 000 files in "Downloads". Why don't you move then to their appropriate folders after downloading? It seems to me like it would be impossible to find anything on that desktop :wink: 

We can't predict how the individual user organizes his/her file system, they could have a lot in their "Pictures" folder they don't want to share with their wife as well. That's why I think the best option would be to let them choose if they want to share the default folders or simply share nothing during the wizard.

I think that with defaults in general (not just default folders), we can never make it fit everyone. That's why they are just defaults, and can be changed. When setting defaults we have to try to estimate what would be the best choice for the majority of users, and for those that don't understand file systems I still think including "Desktop" and "Downloads" will be helpful. 

I'd also like to emphasize that normally when we talk about privacy on computers, we talk about what can end up in the hands of others on the Internet. DMS doesn't make anything available on the Internet, it will only share *media files*, and only on the *local network*. The reason for this is that it relies on multicast, which isn't routed by default in most routers, and which certainly isn't routed by ISP routers. 

That means that even if a user have no router and no firewall, DMS will only work on the local network. It is however possible that a user could configure their router with port forwarding to the web interface, and make it available to others on the Internet that way, but then we're not talking about the novice users that don't know what they share.

**Comment by onon765trb** [comment]
@Sami32
You are right. At the same, I think developers should not help leaving users danger by defaults, knowing it.  It is "矜持" (kyouji, Japanese) as a developer. It means all of pride, soul, spirit, self-esteem, self-control. Translation is difficult.
But, if users share any folders by their willings, it is their responsibilities. What developers can do are only warning.
defaults means developers approve, recommend and take responsibility of  the setting. Even if developers do not think so, users and  the public eye think so.
ExSport says sharing all drives is **stupid** in a thread. You can find it easily. I stop to use a link for avoiding confusion.

@Nadahar
>Why don't you move then to their appropriate

Busy.Impossible without being very meticulous. Spotlight is very useful. Those who move files  to their appropriate folder at everytime is minor, even Japanese.

>I still think including "Desktop" and "Downloads" will be helpful.

I understand it very very very well. I think responsibilities are not only " helpful" or "useful".

Is what i say difficult?

**Comment by Sami32** [comment]
@onon765trb Developpers should not consider the users as idiots IMO!
As this sofware is for them and that we cannot satisfy everyone, the democratic way should be prefered.
So we try to listen their real complain, worries and do compromise to satisfy the most of them. 
I wish that democratie should be more widely shared around the world.

**Comment by Nadahar** [comment]
@onon765trb 
> Busy. Impossible without being very meticulous

We must have very different habits, I generally don't have any problem with moving files that matter to the appropriate location. Sure, I have some temporary files and other non-important stuff on my desktop which I clean up from time to time, but I never keep important stuff on the desktop. That means that I don't worry at all if my desktop content is shared with others in the house :wink: 

> Is what i say difficult?

It's not difficult to remove the folders you mention from the selection of default folders. For me, this is about making the correct decision. There's no rush, it's all about making the correct decision.

I understand your view about default although I disagree to some degree. I know that the attitude among Mac users is different that of users of other platforms. I think Apple is to blame for this, because they have taught users to be "lazy" and use "their" settings instead of taking responsibility themselves. It all has to do with Apples idea of "user friendliness", which means little or no choice. It is generally accepted in some circles that removing settings (choices) and relying only on defaults is somehow user friendly, and sadly this idea has also spread to Android and partly to the latest versions of Windows. I don't think this is user friendly at all, I think it's dumbing things down. It is "user friendly" only for those that are easily scared by having to make choices, and for which the defaults work well. It's simply frustrating for anyone that wants to have control of how things work or where the defaults doesn't work. In short, I don't believe in the "Apple way", and don't want to copy it, even for macOS.

Instead I believe in exposing different "difficulties" for users of different levels. I have long been planning to have different "levels" for the GUI to adjust to the "level" of the user, something like: Apple, Basic, Normal, Advanced, Expert, Developer. This would then be the first thing the user would choose, and everything happening since then would depend on that choice.

This is a to a large degree a digression here, but I wanted to give some more context to my views. What the above does mean though, is that I probably have a slightly different view of what a default should be.

When a certain default will work the best for almost all users, it's an easy choice. It's situations like this, where the default can't possibly "fit" for everyone, that it gets more difficult. That means that we have to apply principles when we decide which default to choose. I think you and I have slightly different principles that applies, or that the "weight" of the principles differ.

To sum up my "calculation":
* Privacy and security weighs heavily, but I don't see sharing something on the local network (with your wife/kids) as much of a privacy or security concern.
* I want to find the settings that I think will work well for the largest portion of users. In this case I don't expect anybody than the most novice users to actually use the default shares, and as such I worry most about what would be helpful for them.
* I want to find a setting that is reasonable logical/intuitive, but that is highly subjective in this case so it must largely be disregarded.

**Comment by Nadahar** [comment]
I think I should make a suggestion of the solution I think of, to make sure that we're all discussing the same solution :relaxed:

**Comment by onon765trb** [comment]
@Sami32 
>the democratic way should be prefered.

I agree with it, greatly. At the same time, we should not forget that the democratic way makes Nazi under the most democratic  constitution in history, as you know. **Stupid**
This teach us the way of operation and intelligence assessments are important. 
EG. most of people do not know safe regulations on TV or BD/DVD players.
I have the proof of self-made performance by a developper. 

@Nadahar 
The default increases the burden of PC and the risks and decreases reputation. The reputation of the previous version of DMS is not good.
The folder instantly appears by only one click on add button.

What are advantages to save the one click in the situation?

**Comment by Sami32** [comment]
@onon765trb My point was that after having read real users reports you understand that someone will have to help, answer them. Who?
My math was simple: the time used for this task will not be used to fix and enhance our code.

Note about our metaphoric  disgression ;) 
 I never considered Germany as having the democratic constitution but ok.
I guess that Trump should also stay in history books to show how all our "democraties" are flawed and manipulated by lobbies and groups of interest. 
I consider that japanese in Asia and frenches in Algeria have done as bad as the Nazis.
About "stupid" word, i always find this funny and dangerous, as we are all ignorant at some point and what we can see as stupid could be seen differently by other.
We always end to be the stupid guy for someone anyways ;-) 

Please don't misunderstand me, i'm not saying that i found doing that is the way to go, just that i can understand why many users do that this way. 
That was my 2 cents.

**Comment by Nadahar** [comment]
@onon765trb I'm not sure I understand what you mean. I have explained why I think it's important that new users get something to work without too much trouble. At the same time my suggestion in #107 should make it easy for those that have things to hide to avoid it being shared with their family.

I don't understand what you mean by "only one click on add button". As I've written before, I know from experience that many users don't understand file systems, and for them a "folder" is an unknown concept and browsing to where they have their files is likely to be an impossible task.

**Comment by onon765trb** [comment]
@Sami32 @Nadahar
I cannot think it is advantage and help. It leaves users danger and high burden. They increases for beginners, e.g. users do not understand file systems. There is the possibility of unexpected sharing by mishandling reset to default button.
All the more, even if such requests are real, it should be not accepted. Shared folders should be selected by users self as most servers do so.
DMS is cool IF, very stable, fast responses and perfect free. On the other hand, other servers are...
It means DMS is a a threat for them because they gain from denotes, ad. fee etc. They wait bad reputations of DMS.
But I have given up. I cannot recommend the present DMS to friends. Based on deep consideration, I cannot cooperate the present DMS with great regret because of forcing danger and high burden though I complete many PR for improvements of transcoding.
It seems a developer wish this.
But I will support as a user. I may PR when the policy of DMS changes.

Thank you and see you!

**Comment by Nadahar** [comment]
@onon765trb I'm sorry you feel that way. As I've tried to say before, we always have to try to balance the need of different users when selecting defaults. There will always be compromises that are less than perfect, but our goal is that it should work as well as possible for as many as possible.

To me it feels like you have simply decided that "default folders" is bad and can't exist, and that your arguments and goals keep changing as solutions are proposed.

As I've stated before, I don't even agree that there's a privacy issue here. DMS will only share media on the local network, nothing is shared on the internet. The only "privacy" one has to worry about as such is with regards to others on the local network, which will normally be your close family. I have a hard time to see that it's a common problem if your family has access to your media files. In fact, I think that's why many people will choose to run a media server. I think characterizing this as a "great danger" is very exaggerated.

The firewall/OS can be used to control access to DMS like any other application, in addition to DMS' IP filtering. I can't see that there's any "risk" that the "default folders" button is pressed by accident, a confirmation dialog is shown (I think, if it's not, that should be added) if manual folders are defined.

I don't think there's an unreasonable burden on users, users that don't understand filesystems won't be able to add shared folders manually, so their only option to get this working *is* to use the default folders.

I can't speak for all media servers out there, but Windows itself will share the standard media folders by default if you join a "home group", which the default "wizard" encourages you to do.

In addition to that, I'm planning another system for controlling renderer access in the future, where you can select to block all by default.

I'm open to solution that will protect people from accidental sharing as long as it retains the ability for beginners to get the system up and running easily. I think that the wast majority of users don't have any media files that they hide from those they live with, and I don't expect that those who do keep these in the "standard" folders. In addition I think it's easy for those few that has media files in the "standard folders" they can't share with the others on the local network to avoid sharing these.

**Comment by Sami32** [comment]
@onon765trb I respect your choice :-(
I couldn't say better than @nadahar with my poor english but i share his words. We just try honestly to make most users experience easy and safe. And sometimes the balance can be tricky to get.
As you already saw we are not the ones running after hype and ads income ;)
Only after feedback and testers :wink:
So, please feel free to share suggestions, features or issue reports and test anytime :-D
